src/js/simpleSmtpServer.js

```diff
@@ -0,0 +1,84 @@
+// A module to wire in simplesmtp to handle incoming messages.
+"use strict";
+var fluid = fluid || require("infusion");
+var gpii  = fluid.registerNamespace("gpii");
+fluid.registerNamespace("gpii.test.mail.smtp.simpleSmtpServer");
+
+var simplesmtp = require("simplesmtp");
+var fs         = require("fs");
+
+gpii.test.mail.smtp.simpleSmtpServer.handleStartData = function (that, connection) {
+    var timestamp = (new Date()).getTime();
+    that.options.messageFile = that.options.config.outputDir + "/message-" + timestamp + ".txt";
+    connection.saveStream = fs.createWriteStream(that.options.messageFile);
+};
+
+gpii.test.mail.smtp.simpleSmtpServer.handleData = function (that, connection, chunk) {
+    connection.saveStream.write(chunk);
+};
+
+gpii.test.mail.smtp.simpleSmtpServer.handleDataReady = function (that, connection, callback) {
+    connection.saveStream.end();
+
+    that.events.messageReceived.fire(that, connection);
+
+    callback(null, that.options.config.queueId);
```

Comment:
Yes, in looking at this I can see that the callback is a bad pattern.  I'm also considering having a separate named event that is meant to be bound to instead of the raw events that mirror simplesmtp's lifecycle.